cmd/dagger/shell_exec.go

```diff
@@ -678,11 +678,13 @@ func (h *shellCallHandler) GetDependency(ctx context.Context, name string) (*She
 		return nil, nil, fmt.Errorf("dependency %q not found", name)
 	}
 	st, err := h.getOrInitDefState(dep.ModRef, func() (*moduleDef, error) {
-		var opts []dagger.ModuleSourceOpts
+		opt := dagger.ModuleSourceOpts{
+			DisableFindUp: true,
+		}
 		if dep.RefPin != "" {
-			opts = append(opts, dagger.ModuleSourceOpts{RefPin: dep.RefPin})
+			opt.RefPin = dep.RefPin
 		}
-		return initializeModule(ctx, h.dag, dep.ModRef, false, opts...)
+		return initializeModule(ctx, h.dag, dep.ModRef, opt)
```

Comment:
It was silly of me to add the `if dep.RefPin != ""` condition 😅 because empty value options are stripped from the request anyway. Assuming the rename of `initializeModuleConfig` to `initializeModule`, all of this becomes:
```go
	return initializeModule(ctx, h.dag, h.dag.ModuleSource(dep.ModRef, dagger.ModuleSourceOpts{
		DisableFindUp: true,
		RefPin: dep.RefPin,
	})
```
Or add a method to `moduleDependency`:
```go
	return initializeModule(ctx, h.dag, dep.Source())
```

Reply:
Simplified this

Reply:
Missing the `RefPin`, but I pushed it to your branch.